src/reducers/entities/search.js

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {combineReducers} from 'redux';
+import {SearchTypes, UserTypes} from 'action_types';
+
+function handleReceivedSearch(data) {
+    const newPosts = data.posts;
+    const postsForSearch = Object.values(newPosts);
+
+    // Sort to ensure that the most recent posts are first
+    postsForSearch.sort((a, b) => {
+        if (a.create_at > b.create_at) {
+            return -1;
+        } else if (a.create_at < b.create_at) {
+            return 1;
+        }
+
+        return 0;
+    });
+
+    return postsForSearch.map((post) => post.id);
+}
+
+function results(state = [], action) {
+    switch (action.type) {
+    case SearchTypes.RECEIVED_SEARCH_POSTS: {
+        return handleReceivedSearch(action.data);
+    }
+    case SearchTypes.REMOVE_SEARCH_POSTS:
+    case UserTypes.LOGOUT_SUCCESS:
+        return [];
+
+    default:
+        return state;
+    }
+}
+
+function recent(state = {}, action) {
+    const nextState = {...state};
+    const {data, type} = action;
+    switch (type) {
+    case SearchTypes.RECEIVED_SEARCH_TERM: {
+        const team = {
```

Comment:
Javascript [doesn't guarantee key ordering](https://stackoverflow.com/a/5525820), so I don't think we should rely on that.

And for a cap, I'm more meaning a hard cap so that we don't store unbounded amounts of search results here, but this particular case isn't much of a problem since we don't limit that elsewhere in the redux store anyway, and that is storing way more data :p